compliance_suite/job_runner.py

```diff
@@ -73,16 +76,16 @@ def generate_summary(self) -> None:
         failed_tests_count: int = len(self.test_status["failed"])
         skipped_tests_count: int = len(self.test_status["skipped"])
 
-        logger.log(LOGGING_LEVEL['SUMMARY'], "\n\n\n")
-        logger.log(LOGGING_LEVEL['SUMMARY'], "{:#^90}".format("   Compliance Testing Summary   "))
-        logger.log(LOGGING_LEVEL['SUMMARY'], "#{:^88}#".format(""))
-        logger.log(LOGGING_LEVEL['SUMMARY'], "#{:^88}#".format(f"Total Tests - {self.test_count}"))
-        logger.log(LOGGING_LEVEL['SUMMARY'], "#{:^88}#".format(f'Passed - {passed_tests_count} ({passed_tests})'))
-        logger.log(LOGGING_LEVEL['SUMMARY'], "#{:^88}#".format(f'Failed - {failed_tests_count} ({failed_tests})'))
-        logger.log(LOGGING_LEVEL['SUMMARY'], "#{:^88}#".format(f'Skipped - {skipped_tests_count} ({skipped_tests})'))
-        logger.log(LOGGING_LEVEL['SUMMARY'], "#{:^88}#".format(""))
-        logger.log(LOGGING_LEVEL['SUMMARY'], "{:#^90}".format(""))
-        logger.log(LOGGING_LEVEL['SUMMARY'], "\n\n\n")
+        logger.summary("\n\n\n")
+        logger.summary(PATTERN_HASH_CENTERED.format("   Compliance Testing Summary   "))
+        logger.summary(PATTERN_HASH_SPACED.format(""))
+        logger.summary(PATTERN_HASH_SPACED.format(f"Total Tests - {self.test_count}"))
+        logger.summary(PATTERN_HASH_SPACED.format(f'Passed - {passed_tests_count} ({passed_tests})'))
+        logger.summary(PATTERN_HASH_SPACED.format(f'Failed - {failed_tests_count} ({failed_tests})'))
+        logger.summary(PATTERN_HASH_SPACED.format(f'Skipped - {skipped_tests_count} ({skipped_tests})'))
+        logger.summary(PATTERN_HASH_SPACED.format(""))
+        logger.summary(PATTERN_HASH_CENTERED.format(""))
```

Comment:
Couldn't you write your custom logger class in a way that you can pass the `PATTERN` constants as arguments to the log methods (e.g., `log.error()` and `log.summary()`)? With the complicated syntax and the mix of both `.format()` and f-strings, I think this is (still) a little awkward...